### PR TITLE
Housekeeping github actions

### PR DESCRIPTION
**Pull request by dopey**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -7,62 +7,48 @@ on:
...
-      - name: Checkout
+      -
+        name: Checkout
         uses: actions/checkout@v2
-      - name: Setup Go
+      -
+        name: Setup Go
         uses: actions/setup-go@v2
         with:
           go-version: '^1.15.6'
```
Comment (maraino) [inline .github/workflows/release.yml:25]: Will this use 1.16? Do we want to be backward compatible with older versions of Go?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -175,17 +160,17 @@ jobs:
...
       - name: Checkout
         uses: actions/checkout@v2
       - name: Calculate New SHAs
         id: calculate-shas
         run: |
           PKG=step-cli-bin
-          VVER=${{ needs.create_release.outputs.vversion }}
-          VER=${{ needs.create_release.outputs.version }}
+          VVER=${{ needs.test.outputs.vversion }}
+          VER=${{ needs.test.outputs.version }}
```
Comment (maraino) [inline .github/workflows/release.yml:173]: A few lines below we have 
```
echo "${{ secrets.ARCH_SSH_KEY }}" > /tmp/ssh_arch_key && chmod 0400 /tmp/ssh_arch_key
```
Does this key appear in the logs?

**Review by maraino — COMMENTED** [review]
LGTM but I've added a few questions.

File: .github/workflows/release.yml (hunk 2 above)
Comment (dopey) [reply .github/workflows/release.yml:173]: https://docs.github.com/en/actions/reference/encrypted-secrets#accessing-your-secrets

It shouldn't. This isn't exactly recommended, but I'm open to suggestions.

**Review by maraino — APPROVED** [review]
LGTM, but can se use always the same go-version, right now we're using 1.15.6 and 1.15.8.

Make also sure the oauth commands gets properly merged, I'm not sure why it appears in this PR, it's already in master.

**Pull request merged**